wp/lib/bap_wp/src/precondition.ml

```diff
@@ -895,7 +895,8 @@ let check ?refute:(refute = true) (solver : Z3.Solver.solver) (ctx : Z3.context)
     else
       pre'
   in
-  Z3.Solver.check solver [is_correct]
+  let () = Z3.Solver.add solver [is_correct] in
```

Comment:
What's the difference between running the solver on `is_correct` vs. adding `is_correct` then running the solver?

Reply:
This is the change that made the simple compare test time out vs not time out.
Here's Bjorner's answer about why this is that I don't completely understand
https://github.com/Z3Prover/z3/issues/2654

Reply:
I think more simplifications are run with this new form